lib/storage/index_db.go

```diff
@@ -2744,10 +2744,9 @@ func (is *indexSearch) getMetricIDsForDateTagFilter(tf *tagFilter, date uint64,
 		// Use global search if date isn't set.
 		kb.B = is.marshalCommonPrefix(kb.B[:0], nsPrefixTagToMetricIDs)
 	}
-	kb.B = append(kb.B, tf.prefix[len(commonPrefix):]...)
 	tfNew := *tf
 	tfNew.isNegative = false // isNegative for the original tf is handled by the caller.
-	tfNew.prefix = kb.B
+	tfNew.prefix = append(kb.B, tf.prefix[len(commonPrefix):]...)
```

Comment:
What about returning back this change while passing `commonPrefix` instead of `kb.B` to `tfNew.Init()` at the line 2763?

Reply:
The `commonPrefix` cannot be used instead of `kb.B` at `tfNew.Init()`, since it doesn't contain `{nsPrefixDateTagToMetricIDs, date}` prefix when tag filter for a particular day is used.